File: Atcoder_#352/javascript/b_typing.js
function main(input) {
  const [s, t] = input.trim().split("\n");

  let indexList = [];
  let sIndex = 0;

  for (let i = 0; i < t.length; i++) {
    if (sIndex < s.length && s[sIndex] === t[i]) {
      indexList.push(i + 1);
      sIndex++;
    }
  }

  console.log(indexList.join(" "));
}

main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: Atcoder_#352/ruby/b_typing.rb
s = gets.chomp
t = gets.chomp

index_list = []
s_index = 0

t.length.times do |i|
  if s_index < s.length && s[s_index] == t[i]
    index_list << i + 1
    s_index += 1
  end
end

puts index_list.join(" ")
